### PR TITLE
Add Photo Attachment to Users and Home

### LINKED ISSUE #19: user must upload photo with new home listing

### LINKED ISSUE #23: User can add a profile picture

### LINKED ISSUE #24: User can add photo to home

### PR DESCRIPTION
**Pull request by jwashke**
Implement avatar field in User model
Add S3 file uploads with paperclip to gemfile and config
Implement Photo model with image field
Implement has many photos relationship on homes model
Add photo attachment nested form to new homes form
Implement Validation for Home has at least one Photo
Implement model tests for all above relationships plus photo model

closes #23 
closes #24 
closes #19

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -5,5 +5,8 @@ class User < ActiveRecord::Base
   validates :username,   presence: true, uniqueness: true
   validates :first_name, presence: true
   validates :last_name,  presence: true
+  has_attached_file :avatar,
+  default_url: "https://s3.amazonaws.com/tinystays/avatar-missing.jpeg"
```
Comment (theonlyrao) [inline app/models/user.rb]: nice having default avatar

File: spec/models/photo_spec.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+require 'rails_helper'
+
+RSpec.describe Photo, type: :model do
+  describe "validations" do
+    it { is_expected.to validate_presence_of(:image) }
+  end
+
```
Comment (theonlyrao) [inline spec/models/photo_spec.rb]: nice TDDing the models!

**Comment by theonlyrao** [comment]
This looks good to me. I can also confirm that the tests pass. Patrick I'll let you merge it after you take a look at any possible conflicts in the views.

**Comment by patrickwhardy** [comment]
Yep, looks good to me.

**Pull request merged**